### PR TITLE
feature: domain page

### PR DESCRIPTION
**Pull request by izabellalarsson**
Screenshot 2022-01-14 at 10 48 17

### CODE REVIEW

**Comment by izabellalarsson** [comment]
Screenshot 2022-01-27 at 14 27 42

File: packages/web/src/domains.js [hunk 1]
```diff
@@ -30,60 +54,29 @@ async function getCheck(ctx) {
...
-    const domain = await connection.one(getDomainByID(id))
-    const history = await connection.any(getTestHistoryForDomain(id))
-
-    console.log({domain, history})
-    await ctx.render('domains/show', {domain, history})
+    const domain = await connection.one(getDomainByDomainName(domain_name))
+    const last_test = await connection.maybeOne(getLastTestResultForDomain(domain.id))
+    const history = await connection.any(getTestHistoryForDomain(domain.id, last_test ? [last_test.id] : []))
+
+    console.log({domain, last_test, history})
```
Comment (mull) [inline packages/web/src/domains.js:68]:
```suggestion
```

File: packages/web/src/domains.js [hunk 2]
```diff
@@ -30,60 +54,29 @@ async function getCheck(ctx) {
...
   // TODO: Validate that this domain belongs to this account?
   await ctx.dbPool.connect(async (connection) => {
-    const domain = await connection.one(getDomainByID(id))
-    const history = await connection.any(getTestHistoryForDomain(id))
-
-    console.log({domain, history})
-    await ctx.render('domains/show', {domain, history})
+    const domain = await connection.one(getDomainByDomainName(domain_name))
+    const last_test = await connection.maybeOne(getLastTestResultForDomain(domain.id))
+    const history = await connection.any(getTestHistoryForDomain(domain.id, last_test ? [last_test.id] : []))
```
Comment (mull) [inline packages/web/src/domains.js:66]: If there's no last_test we don't need to make a query at all methinks

**Review by mull — APPROVED** [review]

File: packages/web/src/domains.js (hunk 2 above)
Comment (izabellalarsson) [reply packages/web/src/domains.js:66]: So if last_test don't exist we should not make that query to history at all? That makes sense 😄

**Pull request merged**